sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AppendBlobItem.java

```diff
@@ -55,6 +56,27 @@ public AppendBlobItem(final String eTag, final OffsetDateTime lastModified, fina
     public AppendBlobItem(final String eTag, final OffsetDateTime lastModified, final byte[] contentMd5,
         final boolean isServerEncrypted, final String encryptionKeySha256, final String encryptionScope,
         final String blobAppendOffset, final Integer blobCommittedBlockCount) {
+        this(eTag, lastModified, contentMd5, isServerEncrypted, encryptionKeySha256, encryptionScope, blobAppendOffset,
+            blobCommittedBlockCount, null);
+    }
+
+    /**
+     * Constructs an {@link AppendBlobItem}.
+     *
+     * @param eTag ETag of the append blob.
+     * @param lastModified Last modified time of the append blob.
+     * @param contentMd5 Content MD5 of the append blob.
+     * @param isServerEncrypted Flag indicating if the page blob is encrypted on the server.
+     * @param encryptionKeySha256 The encryption key used to encrypt the append blob.
+     * @param encryptionScope The encryption scope used to encrypt the append blob.
+     * @param blobAppendOffset The offset at which the block was committed to the append blob.
+     * @param blobCommittedBlockCount The number of committed blocks in the append blob.
+     * @param versionId The version identifier of the append blob.
+     */
+    public AppendBlobItem(final String eTag, final OffsetDateTime lastModified, final byte[] contentMd5,
```

Comment:
Mind filing an issue to investigate a way to make constructors which are capable of handling additional parameters that the service could add in the future. There are a few other classes that need a new constructor every time a new service return is added.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/9530